File: LeetCode/Python/leetcode-3634-minimum-removals-to-balance-array.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        n = len(nums)
        i = j = 0
        nums.sort()
        mx = mn = nums[0]
        l = 1

        while j<n:
            mx = nums[j]
            mn = nums[i]

            while i<j and mx > k*mn:
                i += 1
                mn = nums[i]
            
            l = max(l,j-i+1)
            j += 1
        
        return n-l

        

File: LeetCode/C++/leetcode_3634_minimum-removals-to-balance-array.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        int n = nums.size();
        if(n==1) return 0;
        int mx = nums[0];
        int mn = nums[0];
        int l = 1;
        sort(nums.begin(),nums.end());
        int i = 0;
        int j = 0;

        while(j<n){
            mn = nums[i];
            mx = nums[j];
            
            while(i<j and mx>(long long)k*mn){
                i++;
                mn = nums[i];
            }
            l = max(l,j-i+1);
            j++;
        }
        return n-l;
    }
};
